ddl/column.go

```diff
@@ -0,0 +1,266 @@
+		job.SnapshotVer = 0
+		err = t.UpdateTable(schemaID, tblInfo)
+		return errors.Trace(err)
+	case model.StateReorgnization:
+		// reorganization -> public
+		// get the current version for reorgnization if we don't have
+		if job.SnapshotVer == 0 {
+			var ver kv.Version
+			ver, err = d.store.CurrentVersion()
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			job.SnapshotVer = ver.Ver
+		}
+
+		tbl, err := d.getTable(t, schemaID, tblInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		err = d.runReorgJob(func() error {
+			return d.backfillColumn(tbl, columnInfo, job.SnapshotVer)
+		})
+		if errors2.ErrorEqual(err, errWaitReorgTimeout) {
+			// if timeout, we should return, check for the owner and re-wait job done.
+			return nil
+		}
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		// Adjust column offset.
+		d.adjustColumnOffset(tblInfo.Columns, tblInfo.Indices, offset)
+
+		columnInfo.State = model.StatePublic
+
+		if err = t.UpdateTable(schemaID, tblInfo); err != nil {
+			return errors.Trace(err)
+		}
+
+		// finish this job
+		job.SchemaState = model.StatePublic
+		job.State = model.JobDone
+		return nil
+	default:
+		return errors.Errorf("invalid column state %v", columnInfo.State)
+	}
+}
+
+func (d *ddl) onColumnDrop(t *meta.Meta, job *model.Job) error {
+	// TODO: complete it.
+	return nil
+}
+
+func (d *ddl) backfillColumn(t table.Table, columnInfo *model.ColumnInfo, version uint64) error {
+	seekHandle := int64(0)
+	for {
+		handles, err := d.getSnapshotRows(t, version, seekHandle)
+		if err != nil {
+			return errors.Trace(err)
+		} else if len(handles) == 0 {
+			return nil
+		}
+
+		seekHandle = handles[len(handles)-1] + 1
+		// TODO: save seekHandle in reorgnization job, so we can resume this job later from this handle.
+
+		err = d.backfillColumnData(t, columnInfo, handles)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+}
+
+func (d *ddl) backfillColumnData(t table.Table, columnInfo *model.ColumnInfo, handles []int64) error {
+	for _, handle := range handles {
+		log.Info("backfill column...", handle)
+
+		err := kv.RunInNewTxn(d.store, true, func(txn kv.Transaction) error {
+			// First check if row exists.
+			exist, err := checkRowExist(txn, t, handle)
+			if err != nil {
+				return errors.Trace(err)
+			} else if !exist {
+				// If row doesn't exist, skip it.
+				return nil
+			}
+
+			backfillKey := t.RecordKey(handle, &column.Col{ColumnInfo: *columnInfo})
+			_, err = txn.Get(backfillKey)
+			if err != nil && !kv.IsErrNotFound(err) {
+				return errors.Trace(err)
+			}
+
+			// If row column doesn't exist, we need to backfill column.
+			// Lock row first.
+			err = txn.LockKeys(t.RecordKey(handle, nil))
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			value, _, err := tables.EvalColumnDefaultValue(nil, columnInfo)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			err = t.(*tables.Table).SetColValue(txn, backfillKey, value)
```

Comment:
I suggest export this function in Table interface. 
Otherwise, you must should type conversation, because TableFromMeta may return other type later.